### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by elopez**
This is (very early, ~~untested~~) support for building Echidna in Windows.

Unfortunately `vty` does not support Windows, and as `vty` is a dependency of `brick`, Echidna on Windows does not support interactive mode.

image

### CODE REVIEW

File: lib/Echidna/UI.hs [hunk 1]
```diff
@@ -54,11 +64,18 @@ ui vm world ts d txs = do
       secToUsec = (* 1000000)
       timeoutUsec = secToUsec $ fromMaybe (-1) uiConf.maxTime
       runCampaign = timeout timeoutUsec (campaign updateRef vm world ts d txs)
+#ifdef INTERACTIVE_UI
```
Comment (arcz) [inline lib/Echidna/UI.hs:67]: This is a bit messy but I don't know how to rewrite it in a better way

**Review by arcz — COMMENTED** [review]
Can we have a function

File: lib/Echidna/Solidity.hs [hunk 2]
```diff
@@ -239,7 +244,11 @@ loadSpecified solConf name cs = do
   where choose []    _        = throwM NoContracts
         choose (c:_) Nothing  = return c
         choose _     (Just n) = maybe (throwM $ ContractNotFound n) pure $
-                                      find (Data.Text.isSuffixOf (if T.any (== ':') n then n else ":" `append` n) . (.contractName)) cs
+                                      find (Data.Text.isSuffixOf (contractId n) . (.contractName)) cs
+        contractId n | T.any (== ':') n = let (splitPath, splitName) = T.breakOn ":" n in
+                                          rewritePathSeparators splitPath `T.append` splitName
+                     | otherwise        = ":" `append` n
+        rewritePathSeparators = T.pack . joinPath . splitDirectories . T.unpack
```
[the remark this answers is not in the document]
Comment (elopez) [reply lib/Echidna/Solidity.hs:251]: solc-returned paths use Windows separators (e.g. a\\b\\c.sol:Foo) so we need to rewrite paths to account for that.

**Review by arcz — APPROVED** [review]

**Comment by rayeaster** [comment]
any idea why I get this error

```
echidna.exe: Couldn't compile given file
stdout:
stderr:
ERROR:CryticCompile:[WinError 2] The system cannot find the file specified
```

**Comment by ggrieco-tob** [comment]
@rayeaster can you please create an issue from this so we can track it down more easily? :slightly_smiling_face:

**Comment by rayeaster** [comment]
> @rayeaster can you please create an issue from this so we can track it down more easily? 🙂

No worry, I get it working like a charm on my local windows now with `crytic-args` option:
```
.\echidna.exe .\testEchidna.sol --contract xxxTest --crytic-args "--solc C:\Users\xxx\Downloads\solcs\solc0810.exe"
```

**Pull request merged**